.github/workflows/spack_cpu_build.yaml

```diff
@@ -18,35 +26,88 @@ jobs:
         # - exago@develop~mpi~ipopt~hiop~python~raja
           # See #44 - +mpi~python should fail if no python in system,
           #                       but the runner happens to have one...
-          - exago@develop+mpi~ipopt+hiop~python~raja ^openmpi
+          - exago@1.6.0+mpi~ipopt+hiop~python~raja ^openmpi
           # See #18 - +hiop~mpi causes issues
         # - exago@develop~mpi~ipopt+hiop~python+raja
           # See #16 - +python~mpi causes issues
         # - exago@develop~mpi~ipopt+hiop+python~raja
-          - exago@develop+mpi~ipopt+hiop+python~raja ^openmpi
+          - exago@1.6.0+mpi~ipopt+hiop+python~raja ^openmpi
           # See #40 - +hiop+raja~ipopt ^hiop~sparse is useful for edge cases
-          - exago@develop+mpi~ipopt+hiop~python+raja ^openmpi ^hiop+raja~sparse 
+          - exago@1.6.0+mpi~ipopt+hiop~python+raja ^openmpi ^hiop+raja~sparse 
 
     name: Build ExaGO with Spack
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
         with:
           # Once we move submodule deps into spack, we can do some more builds
           # Also need to change build script to use spack from base image
           submodules: true
+      
+      - name: Setup Spack
+        run: echo "$PWD/tpl/spack/bin" >> "$GITHUB_PATH"
+      
+      - name: Create heredoc spack.yaml
+        run: |
+          cat << EOF > spack.yaml
+          spack:
+            specs:
+            - ${{ matrix.spack_spec }}
+            concretizer:
+              unify: true
+              reuse: true
+            config:
+              install_tree:
+                root: /opt/spack
+                padded_length: 128
+            packages:
+              all:
+                require: target=x86_64_v2
+            mirrors:
+              local-buildcache: oci://ghcr.io/pnnl/ExaGO
+              spack: https://binaries.spack.io/develop
+
+      - name: Configure GHCR mirror
+        run: spack -e . mirror set --oci-username exago-bot --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache
+      
+      - name: Trust keys
+        run: spack -e . buildcache keys --install --trust
+      
+      - name: Find external packages
+        run: spack -e . external find --all --exclude python
+
+      - name: Concretize
+        run: spack -e . concretize
+  
+      - name: Install
+        run: spack -e . install --no-check-signature
```

Comment:
`-j`?

Reply:
Just letting spack default the number of parallel jobs - I don't think specifying would speed things up, and might cause issues if we suddenly get runners with more/less threads available in future